tools/clang/test/HLSLFileCheck/hlsl/intrinsics/atomic/atomic_restypes.hlsl

```diff
@@ -0,0 +1,131 @@
+// RUN: %dxc -T cs_6_0 -DDEST=ro_structbuf -DIDX=    %s | %FileCheck %s
+// RUN: %dxc -T cs_6_0 -DDEST=ro_structbuf -DIDX=[0] %s | %FileCheck %s
```

Comment:
What is the rule for linking?
Should we report error on foo when compile to lib or only report error when linking with illegal argument a?
export void foo(inout int a, int i) {
  InterlockedAdd(a, i);
}

Reply:
This will fail because it is neither a resource nor a groupshared variable. The only valid entries for atomics currently are resources or groupshared. We have some open questions about passing groupshared parameters to functions that I think are outside the scope of this change. We might provide a way to do this in the future though.

What we can do is pass in resource params, which might be something useful to test.

Reply:
Right, we really can't support atomics on arguments until we support proper references or pointers to the right memory type.